tests/test_mpi.py

```diff
@@ -228,6 +228,148 @@ def test_custom_topology_v2(self, comm_size, topology, dist_topology):
         assert custom_topology == dist_topology
 
 
+class TestSubDistributor:
+
+    sd_specs = [('middle', 2, 1), ('left', 2), ('right', 3)]
+
+    @pytest.mark.parametrize('sd_x', sd_specs)
+    @pytest.mark.parametrize('sd_y', sd_specs)
+    @pytest.mark.parallel(mode=[2])
+    def test_intervals(self, sd_x, sd_y):
+        """
+        Check the interval of indices spanned by the SubDomain is correctly calculated
+        within SubDistributor. Also check that the interval of indices spanned by the
+        SubDomain on each rank is correct.
+        """
+        def check_interval(interval, spec, size, vmin=None, vmax=None):
+            if spec[0] == 'middle':
+                start = spec[1]
+                end = size - spec[2] - 1
+            elif spec[0] == 'left':
+                start = 0
+                end = spec[1] - 1
+            else:
+                start = size - spec[1]
+                end = size - 1
```

Comment:
This can also be parametrized, having an 'expected' argument with the expected to be asserted result

Reply:
I think that would probably make it messier and less extensible, given the combinatorial use of subdomain specifications and their orientation vs the domain decomposition